### ISSUE TITLE
Opensearch serverless image

### ISSUE THREAD

**Issue by gondalez** [opensearch-project/opensearch-build#3992]
### Is your feature request related to a problem? Please describe

It is handy to use the Opensearch docker image for local testing during development to get quick feedback if features will work against AWS Opensearch.

AWS Opensearch _serverless_ recently became available.
It has a restricted set of [api operations](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-genref.html#serverless-operations) and [plugins](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-genref.html#serverless-plugins) when compared to regular opensearch.

Using the (non-serverless) Opensearch docker image will not flush out usages of unsupported operations while in dev.

### Describe the solution you'd like

A docker image that has the same [limitations](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-overview.html#serverless-limitations) as AWS Opensearch Serverless.

### Describe alternatives you've considered

I looked around to see if AWS provide a docker image for serverless, but could not find one.

Apologies if this request doesn't really belong here (since it is not directly affiliated with AWS).
But still, any advice would be appreciated 🙏 :)

### Additional context

\-

**Comment by dblock** [comment]
I think an in image that only includes Serverless-compatible APIs is not a bad idea, however it might be tricky to implement. Serverless, for example, doesn't have the concept of nodes, or has SigV4. I would open a feature request with AWS support either way.

**Comment by gaiksaya** [comment]
> I think an in image that only includes Serverless-compatible APIs is not a bad idea, however it might be tricky to implement. Serverless, for example, doesn't have the concept of nodes, or has SigV4. I would open a feature request with AWS support either way.


Right! Restricting specific APIs should come from underlying software. Docker is one of the distribution created using the product (tarball today). In order to create such image basic requirement would be to have a opensearch-serverless as a separate product.
Moving this request to OpenSearch repository to see if they have any future plans for this. 
Thanks!

cc: @peterzhuamazon  @bbarani @CEHENKLE
